core/server/services/rss/generate-feed.js

```diff
@@ -87,16 +87,17 @@ generateFeed = function generateFeed(baseUrl, data) {
             }
         });
 
-    data.posts.forEach(function forEach(post) {
-        var item = generateItem(post, siteUrl, data.secure);
-
-        filters.doFilter('rss.item', item, post).then(function then(item) {
-            feed.item(item);
+    return data.posts.reduce(function (feedPromise, post) {
+        return feedPromise.then(function () {
+            var item = generateItem(post, siteUrl, data.secure);
```

Comment:
Since we have dropped Node v4, we are making use of `let`/`const`. Could you adapt your code?

BTW: The code works fine now 👍

Reply:
Do you have any specific style guide? E.g. I would also replace regular functions with arrows.

Reply:
They are currently tracked [here](https://github.com/TryGhost/Ghost/issues/9589).

Reply:
Ok, replaced as was said [here](https://github.com/TryGhost/Ghost/pull/9662#discussion_r193425121).